src/main/java/org/traccar/protocol/NavtelecomProtocolDecoder.java

```diff
@@ -308,20 +307,21 @@ protected Object decode(
                                 case 52:
                                     value = buf.readByte();
                                     position.set(
-                                            Position.PREFIX_TEMP + (j + 2 - 45), (value != (byte) 0x80) ? value : null);
+                                            Position.PREFIX_TEMP + (j + 1 - 45), (value != (byte) 0x80) ? value : null);
                                     break;
                                 case 53:
                                     value = buf.readUnsignedShortLE();
                                     if (value != 0x7FFF) {
-                                        if (BitUtil.check(value, 7)) {
-                                            position.set("obdFuelLevel", BitUtil.to(value, 6));
+                                        if (BitUtil.check(value, 15)) {
+                                            position.set("obdFuelLevel", BitUtil.to(value, 14));
                                         } else {
-                                            position.set("obdFuel", BitUtil.to(value, 6) / 10);
+                                            position.set("obdFuel", BitUtil.to(value, 14) / 10.0);
                                         }
                                     }
                                     break;
                                 case 54:
-                                    position.set(Position.KEY_FUEL_USED, buf.readFloatLE());
+                                    double dValue = buf.readFloatLE() * 0.5;
```

Comment:
Why are we excluding zero?

Reply:
For this field value is not valid if < 0

Reply:
OK, can we name the value properly then. I recommend something like `fuelUsed`.

Reply:
Ok!

Reply:
What you did doesn't make sense. It's not even the same line of code. You shouldn't change the key. I'm asking to change `dValue` to `fuelUsed`.

Reply:
done!